packages/pinia/src/devtools/plugin.ts

```diff
@@ -328,6 +328,8 @@ function addStoreToDevtools(app: DevtoolsApp, store: StoreGeneric) {
       store.$onAction(({ after, onError, name, args }) => {
         const groupId = runningActionId++
 
+        const initialState = JSON.parse(JSON.stringify(toRaw(store.$state)));
```

Comment:
We should probably try to use https://developer.mozilla.org/en-US/docs/Web/API/structuredClone if it exists.
It should definitely gracefully fail for circular references and not crash

Reply:
why are you doing `toRaw()` here?

Reply:
`toRaw` helps with cloning the object. I've tested `structuredClone` function but it seems it has problems with cloning functions.

I can create my own deepClone function or use the one that lodash provides, but I don't know if you want to have dependencies

Reply:
there shouldn't be functions in state 😓 
Yes, we want to avoid importing lodash

structuredClone have some limitations but it should also be quite fast. You should probably avoid doing `toRaw()` because we want to unwrap the properties.

I remember Vue devtools had a function for exactly this usecas in their codebase, it might be worth checking it out

Reply:
From what I can tell, because state is wrapped in a proxy with methods, if I don't make this a raw object `structuredClone` won't work.

`JSON.parse(JSON.stringify)` will work without `toRaw` but the effect will be I think the same.

Could you explain to me why `toRaw` should be avoided in this situation?
I use this `raw` object just to copy and display it in devtools.

Reply:
Because of nested refs